### ISSUE TITLE
Parameterize firebase analytics complete_group event

### ISSUE THREAD

**Issue by laurentS** [mapswipe/mapswipe#125]
It just occurred to me that we can send a parameter along with the `complete_group` analytics event, so we could include the area mapped in it, which would give us much more meaningful stats. Think of something like "sale completed" with a "basket value" parameter.

We could even break down per project to do something like:
- `built_area_group_completed` with `km2_mapped` as a parameter. This would give us a chart in firebase with km2 mapped everyday without having to do any extra calculations
- `change_detection_group_completed` with `km2_mapped` for the same results
- `building_footprint_group_completed` with `buildings_validated` for instance.

Or can we even add more than 1 param? so we'd get `total_area_mapped` and `area_with_buildings` for instance?

@Hagellach37 @jhenshall would love your thoughts on this!

**Comment by Hagellach37** [comment]
Hey @laurentS ,
I'm still not sure where the best place for such statistics would be. Firebase Analytics has the limitation that it is visible only to very few people. Mainly project managers. but even not all of them. Adding this functionality in Firebase Analytics would help BRC to get some real-time usage statistics. For example it would be useful to find out, if people will map the new project types (e.g. having an event group completed for "footprint", group completed for "changeDetection). Having those numbers would help us  to see if people "don't use the new project types". Also an event like "BuildAreaTutorial finished" would help us to track if people are actualy using the tutorial, and if not we can try to find out why this is the case. For the information about usage FIrebase Analytics might be a good place, for information about contributions I'm not sure.

I would prefer to have the contribution information more publicly available, e.g. provided by our api. I see the point that currently the api doesn't offer us such a shiny interface as firebase is doing it, but from the information side we might be able to get close it. (and there is already at least something... e.g. if you open this [link](https://apps.mapswipe.org/api/agg_progress_by_project_id_and_date/agg_progress_by_project_id_and_date_-Lq1Bbtgk1YyhES59CeU.csv) you will receive the progress of the BRC project on a daily basis.)

For me the main question is around: Which information do we need to have more useful statistics. And now over to you @jhenshall .

**Comment by jhenshall** [comment]
Love this chat @laurentS @Hagellach37! So, I think we can definitely use parameters better to improve the Firebase analytics, which could help us identify problems in the app or user behaviours of different project types. It would allow us to segment in new ways such as comparing behaviours by operating system, screen size, location, etc. - which could be really interesting.

Agree with @Benni that the final 'contribution stats' should be via the API but is it possible to do both? Firebase can be more saved for the user analysis when we are reviewing the app and how we can improve it. Whereas details on the contributed project data itself always comes from API (and the shiny new interface @Hagellach37 has promised :-). Worth noting that Firebase analytics isn't 100% accurate anyway (more a trend or comparison tool) so the API is the main source of truth anyway.

Would it maybe also be possible to have the project name as a parameter? Could also compare engagement across the different projects and types and the amount mapped?

**Comment by laurentS** [comment]
Yes, we can definitely have both. I was thinking of analytics because adding a parameter is half a line of code, and would give more meaning to the stats we're already looking at, so it's a quick win. It doesn't prevent us at all from building proper APIs for publicly available analyses.

The thought came from looking at the number of groups completed, and realising that I have no idea what 1086 groups represents. But 57 km2 would be more meaningful :)
